tests/test_ble_lifecycle_controller.py

```
@@ -0,0 +1,555 @@
+        # This should use fallback since hook is not callable
+        controller._discard_invalidated_connected_client(
+            mock_client,
+            restore_address=None,
+            restore_last_connection_request=None,
+        )
+
+    def test_fallback_is_closing_calls_lifecycle_service(self) -> None:
+        """_fallback_is_closing should call BLELifecycleService._state_manager_is_closing."""
+        mock_iface = SimpleNamespace()
+        mock_ownership = SimpleNamespace(
+            _discard_invalidated_connected_client=lambda *a, **kw: None
```

Ruff (0.15.6)

[warning] Unused lambda argument: `a`

(ARG005)